src/services/loneAntAnimalGameService.ts

```diff
@@ -0,0 +1,184 @@
+export const TICKS_PER_SECOND = +(process.env.TICKS_PER_SECOND || 10);
+export const MAX_TICKS = +(process.env.MAX_TICKS || 1000);
+const TEST_GRID_WIDTH = 60;
+const TEST_GRID_HEIGHT = 40;
+const TEST_TICKS_PER_SECOND = 10;
+
+export class LoneAntAnimalGameService implements AnimalGameService {
+  loadAnimals(): Animal[] {
+    const path = require('path');
+    const fs = require('fs');
+
+    var ants: Animal[] = [];
+    const files = fs.readdirSync(loneAntsDirectory);
+    files.forEach((file: any) => {
+      const data = fs.readFileSync(path.join(loneAntsDirectory, file));
+      const fileName = path.basename(file, path.extname(file));
+      const username = fileName.substring(0, fileName.indexOf('_'));
+      const antName = fileName.substring(fileName.indexOf('_') + 1);
+      const newAnt = Animal.CreateAnimal(Ant, antName, username, data.toString());
+      ants.push(newAnt);
+    });
+
+    ants = this.shuffle(ants);
+
+    return ants.slice(0, MAX_ANTS);
+  }
+
+  createNewGameBoard() {
+    return this.createGame(
+      this.loadAnimals(),
+      GRID_WIDTH,
+      GRID_HEIGHT,
+      TEST_TICKS_PER_SECOND,
+      MAX_TICKS,
+      FOOD_PERCENTAGE);
+  }
+
+  createTestGameBoard(antName: string, code: string): Game {
+    return this.createGame(
+      [Animal.CreateAnimal(Ant, antName, 'Tester', code)],
+      TEST_GRID_WIDTH,
+      TEST_GRID_HEIGHT,
+      TEST_TICKS_PER_SECOND,
+      MAX_TICKS,
+      FOOD_PERCENTAGE);
+  }
+
+  private createGame(animals: Animal[], width: number, height: number, ticksPerSecond: number, gameLength: number, foodPercentage: number): Game {
+    var board: Board = new Board();
+    // Randomize ant starting position
+    animals.forEach(ant => {
+      ant.row = Math.floor(Math.random() * height);
+      ant.column = Math.floor(Math.random() * width);
+    });
+    board.grid = this.generateGrid(height, width);
+    board.animals = animals;
+    board.food = this.generateFood(height, width, foodPercentage);
+    board.gameStatus =
+    {
+      gameLength: gameLength,
+      foodLeft: board.food.length,
+      elapsedTicks: 0,
+      ticksPerSecond: ticksPerSecond
+    };
+    const game: any = {
+      board,
+      intervalId: undefined
+    };
+
+    game.intervalId = setInterval(() => this.tickGame(game, gameLength), 1000 / ticksPerSecond);
+    return game;
+  }
+
+  // From: https://stackoverflow.com/a/6274381
+  private shuffle(array: any[]) {
+    var j, x, i;
+    for (i = array.length - 1; i > 0; i--) {
+      j = Math.floor(Math.random() * (i + 1));
+      x = array[i];
+      array[i] = array[j];
+      array[j] = x;
+    }
+    return array;
+  }
+
+  generateGrid(height: number, width: number) {
+    const newGrid = Array(height).fill([]);
+    for (var row = 0; row < height; row++) {
+      newGrid[row] = Array(width).fill(1);
+    }
+    return newGrid;
+  }
+
+  generateFood(height: number, width: number, foodPercentage: number) {
```

Comment:
Does it make sense to have generateFood take the height/width/foodPercentage as an argument when they're consts on the class?

Reply:
the constants won't be solely used for long, custom games with different sizes / ticks will be sent in with some new endpoints that are coming